Common/Extensions.cs

```diff
@@ -1445,5 +1445,39 @@ public static string RemoveFromEnd(this string s, string ending)
                 return s;
             }
         }
+
+        /// <summary>
+        /// Queues the specified work to run on the thread pool that must complete within a specified time interval.
+        /// If the work completes within the interval, a new action is invoked if defined.
+        /// </summary>
+        /// <param name="action">The work to execute asynchronously</param>
+        /// <param name="delay">The time span to wait before completing the returned task, or TimeSpan.FromMilliseconds(-1) to wait indefinitely.</param>
+        /// <param name="continuationAction">The work to execute synchronously after the first work is completed</param>
+        /// <exception cref="TimeoutException">The exception that is thrown when the time allotted for the asynchronously work has expired.</exception>
+        public static async Task TimeoutAfterAndContinue(this Action action, TimeSpan? delay, Action continuationAction = null)
+        {
+            using (var timeoutCancellationTokenSource = new CancellationTokenSource())
```

Comment:
You can provide the delay to the `CancellationTokenSource` constructor, and after the delay has passed, it will automatically flag it as cancellation requested. This would also have the benefit of removing the `delayTask`